File: src/main/java/com/rtomyj/yugiohAPI/service/stats/StatsService.java
package com.rtomyj.yugiohAPI.service.stats;

import com.google.inject.internal.util.Strings;
import com.rtomyj.yugiohAPI.dao.database.Dao;
import com.rtomyj.yugiohAPI.model.Stats.DatabaseStats;
import com.rtomyj.yugiohAPI.model.Stats.MonsterTypeStats;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.beans.factory.annotation.Qualifier;
import org.springframework.stereotype.Service;


@Service
public class StatsService {

    private final Dao dao;


    @Autowired
    public StatsService(@Qualifier("jdbc") final Dao dao)
    {

        this.dao = dao;

    }


    public MonsterTypeStats getMonsterTypeStats(final String cardColor)
    {

        final MonsterTypeStats monsterTypeStats = dao.getMonsterTypeStats(Strings.capitalize(cardColor));
        monsterTypeStats.setLinks();

        return monsterTypeStats;

    }


    public DatabaseStats getDatabaseStats()
    {

        final DatabaseStats databaseStats = dao.getDatabaseStats();
        databaseStats.setLinks();

        return databaseStats;

    }

}

File: src/main/kotlin/com/rtomyj/skc/stats/StatsService.kt
package com.rtomyj.skc.stats

import com.rtomyj.skc.dao.StatsDao
import com.rtomyj.skc.exception.ErrorType
import com.rtomyj.skc.exception.SKCException
import com.rtomyj.skc.model.DatabaseStats
import com.rtomyj.skc.model.MonsterTypeStats
import org.slf4j.LoggerFactory
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.beans.factory.annotation.Qualifier
import org.springframework.stereotype.Service

@Service
class StatsService @Autowired constructor(
  @param:Qualifier("jdbc") val dao: StatsDao
) {

  companion object {
    private val log = LoggerFactory.getLogger(this::class.java.name)
  }


  fun getMonsterTypeStats(cardColor: String): MonsterTypeStats {
    val monsterTypeStats = dao.getMonsterTypeStats(cardColor)

    if (monsterTypeStats.monsterTypes.isEmpty()) {
      throw SKCException("Requested monster type not found in DB", ErrorType.DB001)    // flow ends here on err
    }

    log.info("Successfully retrieved stats for monster typing's: {}", monsterTypeStats.toString())
    return monsterTypeStats
  }


  fun databaseStats(): DatabaseStats = dao.getDatabaseStats()
}